crates/fuel-core/src/service/sub_services.rs

```diff
@@ -337,3 +372,97 @@ pub fn init_sub_services(
 
     Ok((services, shared))
 }
+
+fn get_synced_algorithm_updater(
+    config: &Config,
+    genesis_block_height: BlockHeight,
+    settings: ConsensusParametersProvider,
+    block_stream: BoxStream<SharedImportResult>,
+    gas_price_db: Database<GasPriceDatabase, RegularStage<GasPriceDatabase>>,
+    on_chain_db: Database<OnChain, RegularStage<OnChain>>,
+) -> anyhow::Result<
+    FuelGasPriceUpdater<
+        FuelL2BlockSource<ConsensusParametersProvider>,
+        StructuredStorage<Database<GasPriceDatabase, RegularStage<GasPriceDatabase>>>,
+    >,
+> {
+    // if gas price metadata height is behind the latest block height,
+    // we should create an updater and manually update the gas price by adding the missing l2 blocks
+    // from the on_chain database
+    let metadata_height: u32 = gas_price_db
+        .latest_height()?
+        .unwrap_or(genesis_block_height)
+        .into();
+    let latest_block_height: u32 = on_chain_db
+        .latest_height()?
+        .unwrap_or(genesis_block_height)
+        .into();
+    let default_metadata = UpdaterMetadata::V0(V0Metadata {
+        new_exec_price: config.starting_gas_price,
+        min_exec_gas_price: config.min_gas_price,
+        exec_gas_price_change_percent: config.gas_price_change_percent,
+        l2_block_height: genesis_block_height.into(),
+        l2_block_fullness_threshold_percent: config.gas_price_threshold_percent,
+    });
+    if metadata_height < latest_block_height {
+        let mut metadata_storage = StructuredStorage::new(gas_price_db);
+        let metadata = metadata_storage
+            .get_metadata(&metadata_height.into())?
+            .unwrap_or(default_metadata);
+        let mut inner: AlgorithmUpdater = metadata.into();
+        let AlgorithmUpdater::V0(ref mut updater) = &mut inner;
+        for height in (metadata_height + 1)..=latest_block_height {
```

Comment:
This iterates over all the blocks and applies them to the algorithm. This should work for now, but could become confusing when we include the da price calculation in the future.

The confusing part is that the blocks already include the historical gas prices, so do we really need to re-calculate them via the updater?